src/Resources/ResourceManager/Implementation/Policy/SetAzurePolicyAssignment.cs

```diff
@@ -169,6 +182,39 @@ protected override void OnProcessRecord()
             this.WriteObject(this.GetOutputPolicyAssignments(JObject.Parse(result)), enumerateCollection: true);
         }
 
+        /// <summary>
+        /// Verifies if identity type and/or identity ID and/or location are present
+        /// </summary>
+        private bool CheckIfIdentityPresent()
```

Comment:
Please use [AzPSArgumentException](https://github.com/Azure/azure-powershell-common/blob/main/src/Common/Exceptions/AzPSArgumentException.cs) as the exception type in this method. It has more telemetry data to help us analyze, and it's more accurate than PSInvalidOperationException

Reply:
Fixed